data/scripts/eventcallbacks/player/default_onLookInMarket.lua

```diff
@@ -0,0 +1,319 @@
+		end
+
+		-- magic shield (classic)
+		if abilities.manaShield then
+			skillBoosts[#skillBoosts + 1] = "magic shield"
+		end
+
+		-- stats (hp/mp/soul/ml)
+		for stat, value in pairs(abilities.stats) do
+			if value ~= 0 then
+				skillBoosts[#skillBoosts + 1] = string.format("%s %+d", getStatName(stat - 1), value)
+			end
+		end
+
+		-- stats but in %
+		for stat, value in pairs(abilities.statsPercent) do
+			if value ~= 0 then
+				skillBoosts[#skillBoosts + 1] = string.format("%s %+d%%", getStatName(stat - 1), value)
+			end
+		end
+
+		-- speed
+		if abilities.speed ~= 0 then
+			skillBoosts[#skillBoosts + 1] = string.format("speed %+d", math.floor(abilities.speed / 2))
+		end
+
+		-- skills
+		for skill, value in pairs(abilities.skills) do
+			if value ~= 0 then
+				skillBoosts[#skillBoosts + 1] = string.format("%s %+d", getSkillName(skill - 1), value)
+			end
+		end
+
+		-- special skills
+		for skill, value in pairs(abilities.specialSkills) do
+			if value ~= 0 then
+				-- add + symbol to special skill "amount" fields
+				if skill-1 < 6 and skill % 2 == 1 then
+					value = string.format("%+d", value)
+				elseif skill-1 >= 6 then
+					-- fatal, dodge, momentum coming from the item natively
+					-- (stats coming from tier are near tier info)
+					value = string.format("%0.2f", value/100)
+				end
+
+				skillBoosts[#skillBoosts + 1] = string.format("%s %s%%", getSpecialSkillName(skill-1), value)
+			end
+		end
+
+		-- add to response
+		if #skillBoosts > 0 then
+			response:addString(table.concat(skillBoosts, ", "))
+		else
+			response:addU16(0)
+		end
+	end
+
+	-- charges
+	do
+		if itemType:hasShowCharges() then
+			response:addString(itemType:getCharges())
+		else
+			response:addU16(0)
+		end
+	end
+
+	-- weapon type
+	do
+		if itemType:isWeapon() then
+			response:addString(itemType:getWeaponString())
+		else
+			response:addU16(0)
+		end
+	end
+
+	-- weight
+	response:addString(string.format("%0.2f", itemType:getWeight() / 100))
+
+	-- to do
+	response:addU16(0) -- Imbuement Slots
+	response:addU16(0) -- Magic Shield Capacity
+	response:addU16(0) -- Cleave
+	response:addU16(0) -- Damage Reflection
+	response:addU16(0) -- Perfect Shot
+	response:addU16(0) -- Classification
+	response:addU16(0) -- Tier
+
+	-- buy stats
+	do
+		local stats = itemType:getMarketBuyStatistics(tier)
+		if stats then
+			response:addByte(0x01)
+			response:addU32(stats.numTransactions)
+			response:addU64(stats.totalPrice)
+			response:addU64(stats.highestPrice)
+			response:addU64(stats.lowestPrice)
+		else
+			response:addByte(0x00)
+		end
+	end
+
+	-- sell stats
+	do
+		local stats = itemType:getMarketSellStatistics(tier)
+		if stats then
+			response:addByte(0x01)
+			response:addU32(stats.numTransactions)
+			response:addU64(stats.totalPrice)
+			response:addU64(stats.highestPrice)
+			response:addU64(stats.lowestPrice)
+		else
+			response:addByte(0x00)
+		end
+	end
+
+	response:sendToPlayer(self)
+end
+
+ec:register()
```

Comment:
This file looks double-indented compared to other Lua files, no?

Reply:
`do...end` clauses were used to limit the scope of local variables. This has been done in libs already way before this pull request.

Reply:
why though?

Reply:
In this file it is not necessary to use `do ... end` clauses
There is not a single reason for it

Reply:
I agree the `do .. end` local scope limitation seems pointless. Is the overhead so big that we shouldn't deconstruct them naturally at the end of function, and force it lots of times throughout the execution? Is this some shenanigans to keep memory overhead as low as possible? I'm not to sure how Lua works, but to me it sounds like your making the function slower by calling the garbage collector way too often. 
And indentation size seems to be set to 8, it should be 4.